src/test/kotlin/com/fasterxml/jackson/module/kotlin/test/FailOnNullForPrimitivesTests.kt

```diff
@@ -0,0 +1,52 @@
+package com.fasterxml.jackson.module.kotlin.test
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.JsonMappingException
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import kotlin.test.assertEquals
+
+
+class FailOnNullForPrimitivesTests {
+
+    data class OptionalIntRequiredBoolean(val optInt: Int = -1, val reqBool: Boolean)
+
+    val mapper = jacksonObjectMapper().enable(DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES)
+
+    @Rule
+    @JvmField
+    var thrown: ExpectedException = ExpectedException.none()
+
+    @Test fun `optional primitive parameter defaulted if not in json when FAIL_ON_NULL_FOR_PRIMITIVES is true`() {
+        val actualObj: OptionalIntRequiredBoolean = mapper.readValue("""
+        {
+            "reqBool": false
```

Comment:
Your test title says "if not in JASON" yet it is in the JSON, is this error in the title or in the implementatino?